cloudsync/api.py

```diff
@@ -38,71 +42,109 @@
 )
 
 
-def process_transcode_results(video, job):
+def process_transcode_results(results: dict):
     """
-    Create VideoFile and VideoThumbnail objects for a Video based on AWS ET job output
+    Create VideoFile and VideoThumbnail objects for a Video based on AWS MediaConvert job output.
 
     Args:
-        video(Video): Video object to which files and thumbnails belong
-        job(JSON): JSON representation of AWS ET job output
+        results (dict): The MediaConvert job results from the callback.
     """
+    # Fetch job details
+    video_job = EncodeJob.objects.get(id=results.get("jobId"))
+    video = Video.objects.get(id=video_job.object_id)
+
+    # Update job state
+    video_job.state = 4
+    video_job.message = results
```

Comment:
Can we make sure the result is JSON serializable ? currently it contains datetime objects which are not JSON serializable.
And would you please highlight the duration of the video, we are extracting that while posting video to edX: https://github.com/mitodl/odl-video-service/blob/master/ui/api.py#L200